src/core/qgspointlocator.cpp

```diff
@@ -636,6 +637,9 @@ QgsPointLocator::QgsPointLocator( QgsVectorLayer *layer, const QgsCoordinateRefe
   connect( mLayer, &QgsVectorLayer::featureDeleted, this, &QgsPointLocator::onFeatureDeleted );
   connect( mLayer, &QgsVectorLayer::geometryChanged, this, &QgsPointLocator::onGeometryChanged );
   connect( mLayer, &QgsVectorLayer::dataChanged, this, &QgsPointLocator::destroyIndex );
+  connect( mLayer, &QgsVectorLayer::rendererChanged, this, &QgsPointLocator::destroyIndex );
+  connect( mLayer, &QgsVectorLayer::styleChanged, this, &QgsPointLocator::destroyIndex );
+  connect( mLayer, &QgsVectorLayer::layerModified, this, &QgsPointLocator::destroyIndex );
```

Comment:
Yes, unfortunately, but it's necessary.

Reply:
I have a feeling this may be a blocker: this effectively means that even with every single change of geometry we would be destroying the index - that's going to make digitizing slow... Currently if there is e.g. 1-2 seconds initial lag when the index if first built, such lag would happen after every single move of a node when editing geometries...

Why is it necessary to destroy the index? I would expect we just need to modify existing handlers of featureAdded, featureDeleted, geometryChanged signal handlers to accommodate renderer-dependent search.